src/CLIFramework/Extension/ExtensionBase.php

```diff
@@ -0,0 +1,42 @@
+<?php
+namespace CLIFramework\Extension;
+
+use CLIFramework\ServiceContainer;
+use CLIFramework\Command;
+use CLIFramework\Hook\Hookable;
+use CLIFramework\Hook\HookHolder;
+use CLIFramework\Exception\ExtensionException;
+
+abstract class ExtensionBase
+    implements Extension, Hookable
+{
+    private $hooks;
+
+    private function getHooks()
```

Comment:
Yeah, I don't think we need a `Hookable` interface, since we can register events